core/services/relay/evm/dsl.go

```diff
@@ -0,0 +1,87 @@
+package evm
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	ubig "github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils/big"
+)
+
+type EventFilter struct {
+	Address  common.Address
+	EventSig common.Hash
```

Comment:
LogPoller's `Filter` struct now includes Topic2, Topic3, and Topic4 as well, which Automation LogTriggers needs to pass in.  (Also includes several fields related to retention and rate limiting.)

Somehow we need a way of passing these to down to logpoller's `RegisterFilter()` method, and it can't be coming from the job spec because they need to register and unregister new filters upon user request while the job is running. Not sure if we want to include them here, or find some other way just noting we'll have to incorporate it somehow before LogTriggers starts using ChainReader.

Reply:
There is a separate `EventByIndexFilter`, which is used for filtering topics by value. These are part of the dsl and aren't in any way connected to log poller filters, so this won't be getting passed into` RegisterFilter()`.

Reply:
If I understand how filter re-registering should work, we should be able to just use Bindings like we did for `GetLatestValue` by adding an option unregister/register by a key. This way the product can just do `ReBind(key, address ...)` or just `Bind(key, address ...)`.

Reply:
Ah, I forgot we added `Bind()`!  Yes, I think `Bind()` can play the role of `RegisterFilter()`.  I'm not sure what `ReBind()` is, is that something that we're planning to add?  Why not just call `Bind()` a second time?  Seems like an `Unbind()` might be more useful though... I'm not sure how exactly what you have in mind for unregistering a filter by calling `Bind()` but that seems a bit awkward.